LoopUI/Views/SettingsViewModel.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  SettingsViewModel.swift
+//  LoopUI
+//
+//  Created by Rick Pasetto on 6/25/20.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import Combine
+import LoopCore
+import LoopKit
+import LoopKitUI
+import SwiftUI
+
+public class DeviceViewModel: ObservableObject {
+    public init(deviceManagerUI: DeviceManagerUI.Type? = nil,
+                isSetUp: Bool = true,
+                onTapped: @escaping () -> Void = { }) {
+        self.deviceManagerUI = deviceManagerUI
+        self.isSetUp = isSetUp
+        self.onTapped = onTapped
+    }
+    
+    let deviceManagerUI: DeviceManagerUI.Type?
+
+    @Published private(set) var isSetUp: Bool = false
+    
+    var image: UIImage? { deviceManagerUI?.smallImage }
+    // TODO: Remove the defaults here...they are only here for illustrative purposes
+    var name: String { deviceManagerUI?.name ?? "device" }
+    var details: String { deviceManagerUI?.details ?? "device details"  }
```

Comment:
Any reason to wait on this TODO? Seems like these properties will necessary be optional

Reply:
Strictly speaking, I would like the `DeviceViewModel` contract to be that it will guarantee non-optional name and details (a missing image is fine).  I think I'd rather just keep the fallback string and make them `""`...ok by you?

Reply:
Gotcha.. if that's the case, should `deviceManagerUI` be non-optional in the initializer? If the name and details are non-optional, then it seems to me that concern should be handled at a higher level rather than an implicit empty default, though I could be missing something here